### PR TITLE
Add unit test for odo add binding

### LINKED ISSUE #5627: Create binding between devfile component and existing service
This issue covers the situation when `odo add binding` is executed in a directory with `devfile.yaml`.

#### Select the service instance you want to bind to
List of all objects (CRs) that are bindable. 
This list can be obtained by querying  `bindablekinds`.
`bindable-kinds` gives you just all resources (kinds) that are bindable.
To get the actual object (instances) you need to list objects of a given type

For example using kubectl it is 
```
for r in `kubectl get bindablekinds bindable-kinds -o go-template='{{range .status}}{{.kind}}.{{.group}}{{"\n"}}{{end}}'`; 
do
  kubectl get $r;
done

```

example of how this can be done in go using client-go is in https://github.com/kadel/odo-v3-mockup/blob/c5f12da1c54856b82c431f5173e0086544c015da/cmd/binding.go#L147-L197

#### What will be the ServiceBinding's name?
The name of a new ServiceBinding resource.
The default value should be automatically generated `<devfile name>-<service name>`

#### Bind as files:
Y/N question
sets `bindAsFiles` based on users input
default is Y

### Example:
[asciicast](https://asciinema.org/a/484558)

## non-interactive mode
- [ ] `--service` bindable service to bind to format is `<name>/<kind>.<apigroup>` or just `<name>`.  The command should be smart enough that the user can specify just the name of the object. If there are multiple bindable objects with the same name it errors out and tells user that he/she needs to include the kind and apigroup. Command should validate that the object exists, if not the error message should list what are available options (list all bindable object in a format that user can use with `--service` flag. 
- [ ] `--name` name of newly created service binding object
- [ ] `--bind-as-files` boolean, default is `true` (`--bind-as-files=false` to turn this off)

### PR DESCRIPTION
**Pull request by valaparthvi**
**What type of PR is this:**
/kind tests

**What does this PR do / why we need it:**
This PR adds unit test cases for pkg/binding/binding.go
**Which issue(s) this PR fixes:**

Fixes #5627 

**PR acceptance criteria:**

- [x] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

**Comment by valaparthvi** [comment]
/retest-required
IBM tests pass.

File: pkg/binding/binding_test.go [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+	Version: "v1",
+}
+
+var clusterGVR = schema.GroupVersionResource{
+	Group:    clusterGVK.Group,
+	Version:  clusterGVK.Version,
+	Resource: "clusters",
+}
+
+var deploymentGVR = metav1.GroupVersionResource{
```
Comment (feloy) [inline pkg/binding/binding_test.go:264]: You could use

```
var deploymentGVR = appsv1.SchemeGroupVersion.WithResource("deployments")
```

File: pkg/binding/binding_test.go [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("AddBinding() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+var clusterGVK = schema.GroupVersionKind{
```
Comment (feloy) [inline pkg/binding/binding_test.go:252]: You can first define 
```
var clusterGV = schema.GroupVersion{
	Group:   "postgresql.k8s.enterprisedb.io",
	Version: "v1",
}
```
Then
```
var clusterGVK = clusterGV.WithKind("Cluster")
var clusterGVR = clusterGV.WithResource("clusters")
```

Comment (feloy) [reply pkg/binding/binding_test.go:252]: Could you place these variables at the beginning of the file?

**Pull request merged**